src/detect-http-protocol.c

```diff
@@ -114,6 +114,19 @@ static InspectionBuffer *GetData(DetectEngineThreadCtx *det_ctx,
     return buffer;
 }
 
+static InspectionBuffer *GetData2(DetectEngineThreadCtx *det_ctx,
+        const DetectEngineTransforms *transforms, Flow *_f, const uint8_t _flow_flags, void *txv,
+        const int list_id)
+{
+    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
+    if (buffer->inspect == NULL) {
+        InspectionBufferSetup(det_ctx, list_id, buffer, (const uint8_t *) "HTTP/2", strlen("HTTP/2"));
```

Comment:
maybe the compiler is smart enough to optimize it, but it seems using a runtime length func here is a bit odd.

Reply:
What is the good pattern to do this ?